presto-hive/src/main/java/com/facebook/presto/hive/util/InternalHiveSplitFactory.java

```diff
@@ -197,9 +193,15 @@ private Optional<InternalHiveSplit> createInternalHiveSplit(
             blocks = ImmutableList.of(new InternalHiveBlock(start + length, addresses));
         }
 
-        URI relativePath = partitionInfo.getPath().relativize(path.toUri());
+        String relativePath = path;
+        boolean isRelative = false;
+        if (path.startsWith(partitionInfo.getPath())) {
+            relativePath = path.substring(partitionInfo.getPath().length() + 1);
```

Comment:
Is it always the case it ends with "/"?

Wonder if it is safer to do `path.substring(partitionInfo.getPath().length()` and then simply `partitionInfo.getPath()  + path` in the `InternalHiveSplit#getPath`